include/engine/routing_algorithms/routing_base_ch.hpp

```diff
@@ -382,6 +382,40 @@ inline void search(const datafacade::ContiguousInternalMemoryDataFacade<algorith
            duration_upper_bound);
 }
 
+double getPathDistance(const datafacade::ContiguousInternalMemoryDataFacade<ch::Algorithm> &facade,
+                       const std::vector<NodeID> &packed_path,
+                       const PhantomNode &source_phantom,
+                       const PhantomNode &target_phantom);
+
+// Requires the heaps for be empty
+// If heaps should be adjusted to be initialized outside of this function,
+// the addition of force_loop parameters might be required
+double
+getNetworkDistance(const datafacade::ContiguousInternalMemoryDataFacade<ch::Algorithm> &facade,
+                   SearchEngineData<Algorithm>::QueryHeap &forward_heap,
+                   SearchEngineData<Algorithm>::QueryHeap &reverse_heap,
+                   const PhantomNode &source_phantom,
+                   const PhantomNode &target_phantom,
+                   int duration_upper_bound = INVALID_EDGE_WEIGHT);
```

Comment:
the reason why i did not change is at https://github.com/Project-OSRM/osrm-backend/blob/e568b600f0c4ad5a2cd9b8f97868ef9b1b7fc6b2/src/engine/routing_algorithms/map_matching.cpp#L201-L202  
basically the value is neither weight not duration. It should fixed in #3696